Security Guidelines for New Projects - TAG Security #1022

ragashreeshekar opened the pull request:
Security Guidelines for New Projects aims to provide recommendations for new projects to ensure they follow the minimum measures to secure their projects and build incrementally as their maturity grows. This document is targeted to the maintainers of the project. These guidelines are presented in the following categories:
1. Securing code repositories
2. Self assessment
3. SECURITY.md
4. Incident management
5. Badging

This work is associated with the CNCF TAG Security issue https://github.com/cncf/tag-security/issues/1022 

**Contributors**
Ragashree M C
Chris Aniszczyk

**Reviewers** 
Andrew Martin (CNCF TAG Security Co-Chair)
Brandon Lum (CNCF TAG Security Co-Chair)
Emily Fox (CNCF TOC Security Liaison)
Justin Cormack (CNCF TOC Security Liaison)
Faisal Razzak
Justin Cappos
Marco De Benedictis
Sergey Pronin
Shlomo Zalman Heigh

**Approvers (TAG Security)**
Andrew Martin (CNCF TAG Security Co-chair), Brandon Lum (CNCF TAG Security Co-chair).

**Acknowledgements**
This paper is influenced by the publications of CNCF Technical Advisory Group for Security, particularly the Cloud Native Security Whitepaper[1] and Software Supply Chain Security Best Practices Whitepaper,[2] and CNCF Cloud Native Security Map[3], programs and initiatives by OpenSSF[4][6]. Our sincere gratitude for the contributors of these programs and publications, and to Emily Fox for suggesting the creation of this paper, to the contributors and reviewers of this paper. 

Thanks to @carolynvs for your support towards incorporating this guide into this platform.

carolynvs commented:
@ragashreeshekar I have pushed a change to your branch so that the new security guidelines page is rendered on the website properly. Can you take a look at https://deploy-preview-362--cncf-contribute.netlify.app/maintainers/security/security-guidelines/ to validate that the preview looks as you expected?

I also fixed the footnotes so that they are hyperlinks.

carolynvs commented on website/content/maintainers/security/security-guidelines.md:
```diff
@@ -0,0 +1,241 @@
+
+In addition to scanning your code for identifying vulnerabilities and errors, managing the vulnerabilities dependencies is also important. [Dependabot](https://github.com/dependabot) is one tool that helps in managing vulnerabilities in dependencies by automatically raising a pull request to update vulnerable versions to secure versions of that dependency. These pull requests can be then analyzed ...
+
+For further information on securing the code repository, we recommend reviewing the **GitOps section** of the CNCF Cloud Native Security Whitepaper - [tag-security/cloud-native-security-whitepaper.md at main · cncf/tag-security (github.com)](https://github.com/cncf/tag-security/blob/main/security-whitepaper/v2/cloud-native-security-whitepaper.md#gitopsnew-in-v2) and the **Control Environments** s ...
```
I recommend updating any links that are formatted like this, where the link text is not a title but instead the path in GitHub, to use a more descriptive link text that properly hints to the user where the content will go.

That better matches the style used on the contribute.cncf.io site and is more accessible for people using screen readers.

ragashreeshekar replied:
Addressed in commit [4924939](https://github.com/cncf/tag-contributor-strategy/pull/362/commits/4924939f7c089717b2abc3d461fb867424049fc1)
Please let me know if you need any additional changes. Thanks

carolynvs reviewed (changes requested):
It looks great, I just have a suggestion to improve readability and accessibility.

ragashreeshekar commented:
> @ragashreeshekar I have pushed a change to your branch so that the new security guidelines page is rendered on the website properly. Can you take a look at https://deploy-preview-362--cncf-contribute.netlify.app/maintainers/security/security-guidelines/ to validate that the preview looks as you expected?
> 
> I also fixed the footnotes so that they are hyperlinks.

Thank you very much, this looks great. 
Tiny suggestion for the space in inline references [ x], I will fix this in another commit along with the link text formatting.

caniszczyk commented:
LGTM

jeefy commented on website/content/maintainers/security/security-guidelines.md:
```diff
@@ -0,0 +1,241 @@
+
+### Two Factor Authentication/Multi-factor Authentication
+
+Introducing an additional factor such as "something you have" or "somewhere you are" in the authentication process proves to be of higher effectiveness than relying on one factor "something you know" such as passwords, hence we urge configuring two factor or multi-factor authentication for your accounts, especially for any privileged accounts. For details regarding configuring a 2FA for GitHub, p ...
```
GitHub is beginning to enforce this. It might make sense to reflect that this isn't an urging, but impending requirement by GitHub for any CNCF (or large open source) project. 

See https://github.blog/2023-03-09-raising-the-bar-for-software-security-github-2fa-begins-march-13/ for more info

I do think it's okay to leave as-is, but within a few months the wording should change.

jeefy commented on website/content/maintainers/security/security-guidelines.md:
```diff
@@ -0,0 +1,241 @@
+7. Secure development practices
+8. Resolving security issues
+
+A template to perform the self assessment is available at [CNCF TAG Security Project Resouces - Self-assessment](https://github.com/cncf/tag-security/blob/main/assessments/guide/self-assessment.md). All the assessments (self-assessment and joint assessment) conducted by TAG Security can be found at TAG Security GitHub repository. As an example, self assessments are available within the dedicated  ...
```
(Security.md in particular) -> (SECURITY.md in particular)

nit: Consistency in capitalization

ragashreeshekar replied:
Done.

jeefy commented:
Couple minor nits, otherwise LGTM! Great work!!

TheFoxAtWork commented:
@ragashreeshekar would you take a look at open changes requested from @jeefy ? Once those are resolved and the branch is updated, we should be good to go. Thank you and everyone for all your efforts on this!

ragashreeshekar commented:
> @ragashreeshekar would you take a look at open changes requested from @jeefy ? Once those are resolved and the branch is updated, we should be good to go. Thank you and everyone for all your efforts on this!

I have addressed review comments and updated the branch. 
Thanks @TheFoxAtWork @jeefy @carolynvs for your support in publishing this work.

TheFoxAtWork reviewed (approved):
LGTM !
